lambda_code/src/fhir_repository.py

```diff
@@ -67,33 +63,30 @@ def __init__(self, imms: dict, patient: dict):
         """Create attributes that may be used in dynamodb table"""
         imms_id = imms["id"]
         self.pk = _make_immunization_pk(imms_id)
-        self.patient_pk = _make_patient_pk(
-            [x for x in imms["contained"] if x.get("resourceType") == "Patient"][0][
-                "identifier"
-            ][0]["value"]
-        )
+        if patient:
+            nhs_number = [x for x in imms["contained"] if x.get("resourceType") == "Patient"][0]["identifier"][0][
+                "value"
+            ]
+        else:
+            nhs_number = "TBC"
```

Comment:
Can this logic be tested? e.g. if resource is sent without NHS number is the pk in the dynamoDB table correctly recorded as "TBC"? Some of the other tests in this PR rely on the assumption that this logic works correctly

Reply:
It's awkward because we have no way of getting the PatientPK out of the API. I can see it's going into the dynamoDB table correctly, but no way to evidence it in tests.

Reply:
this is now checked in test_create_immunization.py in TestCreateImmunization.test_no_nhs_number_correct_status